README.md

```diff
@@ -1,2 +1,151 @@
 # stat-fem
+
 Python tools for solving data-constrained finite element problems
+
+[![Build Status](https://travis-ci.com/alan-turing-institute/stat-fem.svg?branch=master)](https://travis-ci.com/alan-turing-institute/stat-fem)
+[![codecov](https://codecov.io/gh/alan-turing-institute/stat-fem/branch/master/graph/badge.svg)](https://codecov.io/gh/alan-turing-institute/stat-fem)
+[![Documentation Status](https://readthedocs.org/projects/stat-fem/badge/?version=latest)](https://stat-fem.readthedocs.io/en/latest/?badge=latest)
+
+## Overview
+
+This package provides a Python implementation of the Statistical Finite Element Method (FEM) as
+described in the paper by Girolami et al. [1] to use data observations to constrain
+FEM models. The package builds on top of Firedrake [2] to assemble the underlying FEM system
+and uses PETSc [3-4] to perform the sparse linear algebra routines. These tools should allow
+the user to create efficient, scalable solvers based on high level Python code to address
+challenging problems in data-driven numerical analysis.
+
+## Installation
+
+### Installing Firedrake
+
+`stat-fem` requires a working Firedrake installation. The easiest way to obtain Firedrake is to
+follow the installation instructions on the [firedrake homepage](https://www.firedrakeproject.org).
+
+### Installing stat-fem
+
+Once you have installed Firedrake, activate the virtual environment that was created as part of
+the installation process. Within the running virtual environment, switch to the main `stat-fem`
+directory and proceed with the installation by entering:
+
+```bash
+$ pip install -r requirements.txt
+$ python setup.py install
+```
+
+This will use `pip` to install any missing dependencies (notably Scipy) and install the `stat-fem`
+package within the Firedrake virtual environment.
+
+### Using a Docker Container
+
+Alternatively, we provide a working Firedrake Docker container that has the `stat-fem` code
+and dependencies installed within the Firedrake virtual environment. See the `docker`
+directory in the `stat-fem` repository.
+
+### Testing the installation
+
+The code comes with a full suite of unit tests. Running the test suite uses pytest and pytest-mpi
+to collect and run the tests. To run the tests on a single process, simply enter `pytest` into
+the running virtual environment from any location in the stat-fem directory. To run the test
+suite in parallel, enter `mpirun -n 2 python -m pytest --with-mpi` or
```

Comment:
Within the docker container, I'm unable to find nor run the tests with `mpirun`. It works when using `mpiexec` though. Was this the case for you too?

Reply:
Ah, yes, good catch. Will fix.